### PR TITLE
Reordering of Marks and Edit Mode

### LINKED ISSUE #2104: Grading Edit Mode
A toggle-able "edit mode". Mentors, TAs and Instructors can only change the gradeable when Edit Mode is enabled. The instructor can disable edit mode for Mentors or TAs and Mentors in the Gradeable.

### PR DESCRIPTION
**Pull request by jchulton**
Closes #2104 and contributes to #1954.
Added the reordering of marks while grading assignments via click and dragging the mark while its component is open
Added "Edit Mode" while grading assignments. While edit mode is disabled, the user will not be able to reorder or change common marks (they will be able to select which marks a student earned). Editing mode can be turned on or off via a checkbox at the top of the grading rubric.

### CODE REVIEW

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 1]
```diff
@@ -1048,17 +1048,25 @@ public function saveSingleComponent() {
...
+                        break;
+                    }
+                }
+
+                $mark->setPoints($post_mark['points']);
+                $mark->setNote($post_mark['note']);
+                $mark->setOrder($post_mark['order']);
                 $mark->save();
-                $_POST['marks'][$index]['selected'] == 'true' ? $mark->setHasMark(true) : $mark->setHasMark(false);
+                $post_mark['selected'] == 'true' ? $mark->setHasMark(true) : $mark->setHasMark(false);
```
Comment (MasterOdin) [inline site/app/controllers/grading/ElectronicGraderController.php]: `$mark->setHasMark($post_mark['selected'] == 'true');` or just `$mark->setHasMark($post_mark['selected'])`?

Comment (CouleeApps) [reply site/app/controllers/grading/ElectronicGraderController.php]: Yeah that makes considerably more sense than this inline-if

**Comment by jchulton** [comment]
Ready for review

**Comment by CouleeApps** [comment]
[Quick demo of what I'm having issues with](https://gfycat.com/ShortDearestBird)

**Pull request closed without merging**